File: server/email-templates.ts
/**
 * Professional email templates for ResidentHive
 */

export interface EmailTemplate {
  subject: string;
  html: string;
  text: string;
}

export class EmailTemplates {
  /**
   * Creates branded email template with consistent styling
   */
  static createTemplate(
    title: string,
    headerColor: string,
    headerText: string,
    content: string,
    ctaButton?: { text: string; url: string; color: string }
  ): string {
    return `
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>${title}</title>
    <style>
        body { font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif; margin: 0; padding: 0; background-color: #f8fafc; }
        .container { max-width: 600px; margin: 0 auto; background-color: white; box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1); }
        .header { background: ${headerColor}; padding: 40px 30px; text-align: center; }
        .logo { color: white; font-size: 32px; font-weight: bold; margin: 0; letter-spacing: -0.5px; }
        .subtitle { color: rgba(255, 255, 255, 0.9); font-size: 16px; margin: 8px 0 0 0; }
        .content { padding: 40px 30px; }
        .cta-button { 
            display: inline-block; 
            background: ${ctaButton?.color || '#3b82f6'}; 
            color: white; 
            text-decoration: none; 
            padding: 16px 32px; 
            border-radius: 8px; 
            font-weight: 600; 
            font-size: 16px; 
            margin: 20px 0;
            transition: all 0.2s ease;
        }
        .cta-button:hover { filter: brightness(1.1); }
        .footer { background-color: #f8fafc; padding: 30px; text-align: center; border-top: 1px solid #e2e8f0; }
        .footer p { color: #64748b; font-size: 14px; margin: 5px 0; }
        .text-center { text-align: center; }
        .mb-4 { margin-bottom: 24px; }
        .text-gray-600 { color: #64748b; }
        .text-gray-800 { color: #1e293b; }
        .font-semibold { font-weight: 600; }
        @media (max-width: 480px) {
            .container { margin: 0 10px; }
            .content { padding: 30px 20px; }
            .header { padding: 30px 20px; }
        }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1 class="logo">🏠 ResidentHive</h1>
            <p class="subtitle">${headerText}</p>
        </div>
        
        <div class="content">
            ${content}
            ${ctaButton ? `
            <div class="text-center">
                <a href="${ctaButton.url}" class="cta-button">${ctaButton.text}</a>
            </div>
            ` : ''}
        </div>
        
        <div class="footer">
            <p><strong>ResidentHive</strong> - AI-Powered Real Estate Platform</p>
            <p>Questions? Contact us at <a href="mailto:support@residenthive.com" style="color: #3b82f6;">support@residenthive.com</a></p>
        </div>
    </div>
</body>
</html>`;
  }

  /**
   * Agent invitation email template
   */
  static getAgentInvitation(agentConfig: { firstName: string; lastName: string; email: string; brokerageName: string }, inviteUrl: string): EmailTemplate {
    const content = `
      <h2 style="color: #1e293b; margin: 0 0 24px 0; font-size: 28px; font-weight: 600;">
        Welcome to ResidentHive, ${agentConfig.firstName}!
      </h2>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        You've been invited to join our AI-powered real estate buyer profile management platform. 
        ResidentHive helps you create intelligent buyer profiles, find perfect property matches, 
        and provide exceptional client experiences.
      </p>
      
      <div style="background-color: #f1f5f9; padding: 24px; border-radius: 8px; margin: 30px 0;">
        <h3 style="color: #1e293b; margin: 0 0 16px 0; font-size: 18px;">Your Account Details</h3>
        <div style="margin: 8px 0; color: #475569;">
          <strong>Name:</strong> ${agentConfig.firstName} ${agentConfig.lastName}
        </div>
        <div style="margin: 8px 0; color: #475569;">
          <strong>Email:</strong> ${agentConfig.email}
        </div>
        <div style="margin: 8px 0; color: #475569;">
          <strong>Brokerage:</strong> ${agentConfig.brokerageName}
        </div>
      </div>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        <strong>Next Steps:</strong><br>
        1. Click the setup button below<br>
        2. Create your secure password<br>
        3. Start creating intelligent buyer profiles<br>
        4. Experience AI-powered property matching
      </p>
      
      <p style="font-size: 14px; color: #64748b; margin: 24px 0 0 0;">
        This invitation link will expire in 7 days for security purposes.
      </p>
    `;

    return {
      subject: 'Welcome to ResidentHive - Complete Your Agent Setup',
      html: this.createTemplate(
        'Welcome to ResidentHive',
        'linear-gradient(135deg, #3b82f6 0%, #1d4ed8 100%)',
        'AI-Powered Real Estate Platform',
        content,
        { text: 'Complete Your Setup', url: inviteUrl, color: 'linear-gradient(135deg, #3b82f6 0%, #1d4ed8 100%)' }
      ),
      text: `Welcome to ResidentHive, ${agentConfig.firstName}!

You've been invited to join our AI-powered real estate buyer profile management platform.

Your Account Details:
- Name: ${agentConfig.firstName} ${agentConfig.lastName}
- Email: ${agentConfig.email}  
- Brokerage: ${agentConfig.brokerageName}

Complete your setup here: ${inviteUrl}

Next Steps:
1. Visit the setup link above
2. Create your secure password
3. Start creating intelligent buyer profiles
4. Experience AI-powered property matching

This invitation link will expire in 7 days for security purposes.

Questions? Contact us at support@residenthive.com
ResidentHive - AI-Powered Real Estate Platform`
    };
  }

  /**
   * Welcome email after account activation
   */
  static getWelcomeEmail(agentConfig: { firstName: string; lastName: string; email: string; brokerageName: string }, dashboardUrl: string): EmailTemplate {
    const content = `
      <h2 style="color: #1e293b; margin: 0 0 24px 0; font-size: 28px; font-weight: 600;">
        Welcome aboard, ${agentConfig.firstName}! 🎉
      </h2>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        Your ResidentHive account is now active and ready to use. You can start creating 
        intelligent buyer profiles and finding perfect property matches for your clients.
      </p>
      
      <div style="background-color: #ecfdf5; padding: 24px; border-radius: 8px; margin: 30px 0; border-left: 4px solid #059669;">
        <h3 style="color: #065f46; margin: 0 0 16px 0; font-size: 18px;">Quick Start Guide</h3>
        <ol style="color: #047857; margin: 0; padding-left: 20px;">
          <li style="margin: 8px 0;">Log in with your email and password</li>
          <li style="margin: 8px 0;">Create your first buyer profile</li>
          <li style="margin: 8px 0;">Experience AI-powered property matching</li>
          <li style="margin: 8px 0;">Share results with your clients</li>
        </ol>
      </div>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        Our AI analyzes buyer preferences, finds perfect matches, and generates professional 
        client presentations - saving you hours while improving client satisfaction.
      </p>
    `;

    return {
      subject: 'Welcome to ResidentHive - Your Account is Ready!',
      html: this.createTemplate(
        'Account Ready',
        'linear-gradient(135deg, #059669 0%, #047857 100%)',
        'Account Successfully Activated!',
        content,
        { text: 'Access Your Dashboard', url: `${dashboardUrl}/agent-login`, color: 'linear-gradient(135deg, #059669 0%, #047857 100%)' }
      ),
      text: `Welcome aboard, ${agentConfig.firstName}!

Your ResidentHive account is now active and ready to use.

Access your dashboard: ${dashboardUrl}/agent-login

Quick Start Guide:
1. Log in with your email and password
2. Create your first buyer profile  
3. Experience AI-powered property matching
4. Share results with your clients

Questions? Contact us at support@residenthive.com
ResidentHive - AI-Powered Real Estate Platform`
    };
  }

  /**
   * Password reset email template
   */
  static getPasswordReset(agentName: string, resetUrl: string): EmailTemplate {
    const content = `
      <h2 style="color: #1e293b; margin: 0 0 24px 0; font-size: 28px; font-weight: 600;">
        Password Reset Request
      </h2>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        Hi ${agentName}, we received a request to reset your ResidentHive password.
      </p>
      
      <p style="font-size: 16px; color: #475569; line-height: 1.6; margin: 0 0 24px 0;">
        If you requested this change, click the button below to set a new password. 
        If you didn't request this, you can safely ignore this email.
      </p>
      
      <p style="font-size: 14px; color: #64748b; margin: 24px 0 0 0;">
        This password reset link will expire in 1 hour for security purposes.
      </p>
    `;

    return {
      subject: 'Reset Your ResidentHive Password',
      html: this.createTemplate(
        'Password Reset',
        'linear-gradient(135deg, #dc2626 0%, #b91c1c 100%)',
        'Password Reset Request',
        content,
        { text: 'Reset Password', url: resetUrl, color: 'linear-gradient(135deg, #dc2626 0%, #b91c1c 100%)' }
      ),
      text: `Password Reset Request

Hi ${agentName}, we received a request to reset your ResidentHive password.

If you requested this change, visit this link to set a new password: ${resetUrl}

If you didn't request this, you can safely ignore this email.

This password reset link will expire in 1 hour for security purposes.

ResidentHive - AI-Powered Real Estate Platform`
    };
  }
}

File: backend/app/services/email_templates.py
"""
HTML email templates for onboarding emails.

Two templates:
1. Broker welcome — sent when admin creates a brokerage
2. Agent invitation — sent when brokerage admin invites an agent
"""


def _base_template(content: str, preview_text: str) -> str:
    """Wrap content in the base email layout."""
    return f"""<!DOCTYPE html>
<html lang="en" xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>ResidenceHive</title>
  <!--[if mso]>
  <noscript>
    <xml>
      <o:OfficeDocumentSettings>
        <o:PixelsPerInch>96</o:PixelsPerInch>
      </o:OfficeDocumentSettings>
    </xml>
  </noscript>
  <![endif]-->
  <style>
    body, table, td, a {{ -webkit-text-size-adjust: 100%; -ms-text-size-adjust: 100%; }}
    table, td {{ mso-table-lspace: 0pt; mso-table-rspace: 0pt; }}
    img {{ -ms-interpolation-mode: bicubic; border: 0; height: auto; line-height: 100%; outline: none; text-decoration: none; }}
    body {{ margin: 0; padding: 0; width: 100% !important; height: 100% !important; background-color: #f4f4f5; font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; }}
    a[x-apple-data-detectors] {{ color: inherit !important; text-decoration: none !important; }}
    @media only screen and (max-width: 600px) {{
      .email-container {{ width: 100% !important; }}
      .px-mobile {{ padding-left: 24px !important; padding-right: 24px !important; }}
    }}
  </style>
</head>
<body style="margin: 0; padding: 0; background-color: #f4f4f5;">
  <div style="display: none; font-size: 1px; color: #f4f4f5; line-height: 1px; max-height: 0px; max-width: 0px; opacity: 0; overflow: hidden;">
    {preview_text}
  </div>
  <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="background-color: #f4f4f5;">
    <tr>
      <td align="center" style="padding: 40px 16px;">
        <!--[if mso]>
        <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="560" align="center"><tr><td>
        <![endif]-->
        <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="max-width: 560px; margin: 0 auto;" class="email-container">
          <!-- Logo -->
          <tr>
            <td align="center" style="padding: 0 0 32px 0;">
              <span style="font-size: 20px; font-weight: 700; color: #18181b; letter-spacing: -0.02em;">&#x1F3E0; ResidenceHive</span>
            </td>
          </tr>
          <!-- Main card -->
          <tr>
            <td>
              <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="background-color: #ffffff; border-radius: 12px; overflow: hidden; box-shadow: 0 1px 3px rgba(0,0,0,0.06);">
                <tr><td style="background-color: #2563eb; height: 4px; font-size: 0; line-height: 0;">&nbsp;</td></tr>
                <tr>
                  <td style="padding: 40px 40px 40px 40px;" class="px-mobile">
                    {content}
                  </td>
                </tr>
              </table>
            </td>
          </tr>
          <!-- Footer -->
          <tr>
            <td style="padding: 28px 0 0 0;">
              <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
                <tr>
                  <td align="center" style="font-size: 12px; color: #a1a1aa; line-height: 1.6;">
                    ResidenceHive Inc.<br>
                    <a href="mailto:piyush@residencehive.com" style="color: #a1a1aa; text-decoration: underline;">piyush@residencehive.com</a>
                  </td>
                </tr>
              </table>
            </td>
          </tr>
        </table>
        <!--[if mso]>
        </td></tr></table>
        <![endif]-->
      </td>
    </tr>
  </table>
</body>
</html>"""


def broker_welcome_email(broker_name: str, brokerage_name: str, signup_url: str) -> str:
    """Generate HTML for the broker welcome email."""
    content = f"""<table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
      <!-- Greeting -->
      <tr>
        <td style="font-size: 22px; font-weight: 700; color: #18181b; line-height: 1.3; padding-bottom: 20px;">
          Your pilot is live, {broker_name}.
        </td>
      </tr>
      <!-- Pain point -->
      <tr>
        <td style="font-size: 15px; color: #3f3f46; line-height: 1.65; padding-bottom: 16px;">
          Every day, leads come in and go cold &mdash; not because your agents aren't working, but because buyers expect an instant, personalized response and generic listing emails don't deliver that.
        </td>
      </tr>
      <tr>
        <td style="font-size: 15px; color: #3f3f46; line-height: 1.65; padding-bottom: 16px;">
          <strong style="color: #18181b;">ResidenceHive changes that from day one.</strong>
        </td>
      </tr>
      <tr>
        <td style="font-size: 15px; color: #3f3f46; line-height: 1.65; padding-bottom: 24px;">
          The moment a lead comes in, your agents get an AI&#8209;powered buyer brief &mdash; personalized to that buyer, backed by live MLS data, and ready to send. No manual work. No delays. Just a response that makes buyers feel understood.
        </td>
      </tr>
      <!-- CTA Button -->
      <tr>
        <td style="padding-bottom: 28px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0">
            <tr>
              <td style="border-radius: 8px; background-color: #2563eb;">
                <a href="{signup_url}" target="_blank" style="display: inline-block; padding: 14px 32px; font-size: 15px; font-weight: 600; color: #ffffff; text-decoration: none; border-radius: 8px;">
                  Activate Your Account &rarr;
                </a>
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Divider -->
      <tr><td style="border-top: 1px solid #e4e4e7; padding-bottom: 24px;"></td></tr>
      <!-- Steps -->
      <tr>
        <td style="font-size: 13px; font-weight: 600; color: #71717a; text-transform: uppercase; letter-spacing: 0.06em; padding-bottom: 16px;">
          Setup takes three steps
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 14px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 13px; font-weight: 700; color: #2563eb; padding-top: 1px;">01</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">Create your account</strong> &mdash; takes 30 seconds.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 14px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 13px; font-weight: 700; color: #2563eb; padding-top: 1px;">02</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">Add your agents</strong> &mdash; just enter their name and email.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 24px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 13px; font-weight: 700; color: #2563eb; padding-top: 1px;">03</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">They're live on the next lead</strong> &mdash; AI buyer briefs start flowing.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Compliance note -->
      <tr>
        <td style="padding-bottom: 24px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="background-color: #f0f9ff; border-radius: 8px; border: 1px solid #bae6fd;">
            <tr>
              <td style="padding: 14px 18px; font-size: 14px; color: #0c4a6e; line-height: 1.5;">
                &#x2696;&#xFE0F; Fair Housing compliance is built in &mdash; your team is protected on every interaction.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Divider -->
      <tr><td style="border-top: 1px solid #e4e4e7; padding-bottom: 24px;"></td></tr>
      <!-- Personal sign-off -->
      <tr>
        <td style="font-size: 15px; color: #3f3f46; line-height: 1.65; padding-bottom: 8px;">
          Let's make this pilot count. Text or call me directly at <strong style="color: #18181b;">(860) 796-9167</strong> &mdash; I want to make sure your team gets value from day one.
        </td>
      </tr>
      <tr>
        <td style="font-size: 15px; color: #18181b; line-height: 1.65; padding-bottom: 4px;">
          <strong>Piyush</strong>
        </td>
      </tr>
      <tr>
        <td style="font-size: 14px; color: #71717a; line-height: 1.5;">
          Founder, ResidenceHive
        </td>
      </tr>
    </table>"""

    return _base_template(
        content,
        f"Your pilot for {brokerage_name} is live. Activate your account and start inviting agents."
    )


def agent_invitation_email(
    agent_name: str,
    broker_name: str,
    brokerage_name: str,
    invitation_url: str,
) -> str:
    """Generate HTML for the agent invitation email."""
    greeting = f"Hi {agent_name}," if agent_name else "Hi there,"

    content = f"""<table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
      <!-- Context banner -->
      <tr>
        <td style="padding-bottom: 20px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="background-color: #eff6ff; border-radius: 8px; border: 1px solid #bfdbfe;">
            <tr>
              <td style="padding: 14px 18px; font-size: 14px; color: #1e40af; line-height: 1.5;">
                <strong>{broker_name}</strong> added you to <strong>{brokerage_name}</strong> on ResidenceHive.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Greeting -->
      <tr>
        <td style="font-size: 22px; font-weight: 700; color: #18181b; line-height: 1.3; padding-bottom: 16px;">
          {greeting}
        </td>
      </tr>
      <!-- Body -->
      <tr>
        <td style="font-size: 15px; color: #3f3f46; line-height: 1.65; padding-bottom: 24px;">
          {brokerage_name} is using ResidenceHive to turn leads into clients faster. Once you join, every new lead gets an AI&#8209;powered buyer brief &mdash; personalized, MLS&#8209;backed, and ready to send. No manual work.
        </td>
      </tr>
      <!-- CTA Button -->
      <tr>
        <td style="padding-bottom: 28px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0">
            <tr>
              <td style="border-radius: 8px; background-color: #2563eb;">
                <a href="{invitation_url}" target="_blank" style="display: inline-block; padding: 14px 32px; font-size: 15px; font-weight: 600; color: #ffffff; text-decoration: none; border-radius: 8px;">
                  Join Your Team &rarr;
                </a>
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Divider -->
      <tr><td style="border-top: 1px solid #e4e4e7; padding-bottom: 24px;"></td></tr>
      <!-- What you'll get -->
      <tr>
        <td style="font-size: 13px; font-weight: 600; color: #71717a; text-transform: uppercase; letter-spacing: 0.06em; padding-bottom: 16px;">
          What you'll get
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 14px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 15px;">&#x1F4CB;</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">AI buyer briefs</strong> with real MLS data, tailored to each lead's budget and preferences.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 14px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 15px;">&#x1F4F1;</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">WhatsApp &amp; iMessage</strong> &mdash; manage leads and send briefs right from your phone.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <tr>
        <td style="padding-bottom: 24px;">
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
            <tr>
              <td width="32" valign="top" style="font-size: 15px;">&#x2696;&#xFE0F;</td>
              <td style="font-size: 15px; color: #3f3f46; line-height: 1.55;">
                <strong style="color: #18181b;">Built&#8209;in compliance</strong> &mdash; Fair Housing flags and disclosure tracking handled for you.
              </td>
            </tr>
          </table>
        </td>
      </tr>
      <!-- Expiry -->
      <tr>
        <td>
          <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%" style="background-color: #fafafa; border-radius: 8px;">
            <tr>
              <td style="padding: 14px 18px; font-size: 13px; color: #71717a; line-height: 1.5;">
                Setup takes 2 minutes. This invitation expires in 7 days.
              </td>
            </tr>
          </table>
        </td>
      </tr>
    </table>"""

    return _base_template(
        content,
        f"{broker_name} added you to {brokerage_name} on ResidenceHive. Join your team."
    )
